utilities/claim/claim.js

```diff
@@ -160,49 +168,64 @@ async function initiateRound(privateKey, { ethRegistryAddress, ethTokenAddress,
-        );
-        await new Promise((resolve) => setTimeout(resolve, 1000))
-        vaa = await wh.getVaa(whm, "TokenBridge:Transfer", oneHourInMs)
-      }
-      const isComplete = await tokenBridge.isTransferCompleted(vaa)
-      console.log(`Wormhole redemption completion status: ${isComplete}`)
-      // Wait for 30 seconds to ensure the VAA is processed
-      await new Promise((resolve) => setTimeout(resolve, 30_000))
-      if (!isComplete) {
-        const redeemTxs = tokenBridge.redeem(signer.address.address, vaa)
-        const resRedeem = await signSendWait(
-          receiveChain,
-          redeemTxs,
-          signer.signer
-        )
-      }
-      console.log('Wormhole redemption completed')
-
+    // Read and process all pending transactions
+    const transactions = fs.readFileSync(wormholeFile, 'utf8').split('\n').filter(Boolean)
+    let remainingTxs = transactions
+    if (transactions.length === 0) {
+      console.log('No transactions to redeem')
     } else {
-      console.log('No value to transfer from EthRewardsManager')
+      for (const pendingTxHash of transactions) {
+        console.log('Redeeming wormhole funds on Solana for transaction:', pendingTxHash)
+        const wh = await wormhole("Mainnet", [evm, solana])
+        const sendChain = wh.getChain("Ethereum")
+        const receiveChain = wh.getChain("Solana") 
+        const tokenBridge = await receiveChain.getTokenBridge()
+        const signer = await getWormholeSolanaSigner(receiveChain, feePayerSecretKey, solanaRpcEndpoint)
+        const [whm] = await sendChain.parseTransaction(pendingTxHash)
+    
+        const oneHourInMs = 3600000
+    
+        let vaa = await wh.getVaa(whm, "TokenBridge:Transfer", oneHourInMs)
+        let retryCount = 0
+        while (!vaa) {
+          logger.warn(
+            { tx: pendingTxHash, retryCount: retryCount++ },
+            "No TokenBridgeTransfer VAA found, waiting and trying again..."
+          );
+          await new Promise((resolve) => setTimeout(resolve, 1000))
+          vaa = await wh.getVaa(whm, "TokenBridge:Transfer", oneHourInMs)
+        }
+        const isComplete = await tokenBridge.isTransferCompleted(vaa)
+        console.log(`Wormhole redemption completion status: ${isComplete}`)
+        // Wait for 30 seconds to ensure the VAA is processed
+        await new Promise((resolve) => setTimeout(resolve, 30_000))
+        if (!isComplete) {
+          const redeemTxs = tokenBridge.redeem(signer.address.address, vaa)
+          const resRedeem = await signSendWait(
+            receiveChain,
+            redeemTxs,
+            signer.signer
+          )
+        }
+        console.log('Wormhole redemption completed')
+  
+        // Remove completed transaction from file
+        remainingTxs = remainingTxs.filter(tx => tx !== pendingTxHash)
+      }
+      fs.writeFileSync(wormholeFile, remainingTxs.join('\n') + (remainingTxs.length ? '\n' : ''))
```

Comment:
Sort of similar concern. If we crash on one of the middle txs, we would not end up removing any successfully processed txs from the input file and would try them again. If that's safe, then maybe not a big deal. But maybe a `finally` here that guarantees we write the remaining txs to file would be a good idea?

Reply:
It's safe enough, not really concerned. they'd get immediately returned as completed